integrations/test-gitlab-v2/client.py

```diff
@@ -0,0 +1,105 @@
+import httpx
+from typing import List, Dict, Any, AsyncIterator
+from loguru import logger
+from port_ocean.context.ocean import ocean
+
+class GitLabClient:
+    def __init__(self, base_url: str, token: str):
+        self.base_url = base_url
+        self.token = token
+        self.headers = {"Authorization": f"Bearer {self.token}"}
+        logger.info(f"GitLabClient initialized with base_url: {self.base_url}")
+
+    async def _request(self, method: str, endpoint: str, **kwargs) -> dict:
+        url = f"{self.base_url}/{endpoint}"
+        logger.info(f"Making {method} request to {url}")
+        async with httpx.AsyncClient() as client:
+            response = await client.request(method, url, headers=self.headers, timeout=60, **kwargs)
+            logger.info(f"Response received from {url} with status code {response.status_code}")
+            response.raise_for_status()  # Raise error for non-2xx status
+            return response.json()
+
+    async def get_groups(self) -> AsyncIterator[List[Dict[str, Any]]]:
+        """Fetch groups from GitLab."""
+        endpoint = "groups"
+        logger.info("Fetching groups from GitLab")
+        async for page in self._paginated_request("GET", endpoint):
+            logger.info(f"Fetched page with {len(page)} groups")
+            yield page
+
+    async def get_merge_requests(self) -> AsyncIterator[List[Dict[str, Any]]]:
+        """Fetch merge requests from GitLab."""
+        endpoint = "merge_requests"
+        logger.info("Fetching merge requests from GitLab")
+        async for page in self._paginated_request("GET", endpoint):
+            logger.info(f"Fetched page with {len(page)} merge requests")
+            yield page
+
+    async def get_issues(self) -> AsyncIterator[List[Dict[str, Any]]]:
+        """Fetch issues from GitLab."""
+        endpoint = "issues"
+        logger.info("Fetching issues from GitLab")
+        async for page in self._paginated_request("GET", endpoint):
+            logger.info(f"Fetched page with {len(page)} issues")
+            yield page
+
+    async def get_projects(self) -> AsyncIterator[List[Dict[str, Any]]]:
+        """Fetch projects from GitLab."""
+        endpoint = "projects"
+        logger.info("Fetching projects from GitLab")
+        async for page in self._paginated_request("GET", endpoint):
+            logger.info(f"Fetched page with {len(page)} projects")
+            yield page
```

Comment:
you don't need to have a dedicated method for each kind. since the API is well structured, you could have just one function, such as get_paginated_resource(endpoint) which will be responsible to get the data from GitLab